osdep/timer-linux.c

```diff
@@ -38,7 +38,7 @@ uint64_t mp_raw_time_ns(void)
 #if defined(CLOCK_MONOTONIC_RAW)
     clock_gettime(CLOCK_MONOTONIC_RAW, &tp);
 #else
-    timespec_get(&tp, TIME_UTC);
+    clock_gettime(CLOCK_MONOTONIC, &tp);
```

Comment:
Fine with me, if we always require `clock_gettime` to be present. It wasn't the case before. But apparently we dropped support for any system that doesn't have that.

But this probably should be runtime fallback in this case. 

```suggestion
    int ret = EINVAL;
#if defined(CLOCK_MONOTONIC_RAW)
    ret = clock_gettime(CLOCK_MONOTONIC, &tp);
#endif

    if (ret == EINVAL)
        ret = clock_gettime(CLOCK_MONOTONIC, &tp);
        
    if (ret == EINVAL)
        timespec_get(&tp, TIME_UTC);
```

`timespec_get(&tp, TIME_UTC);` is optional, but might as well do it...

Reply:
Well `TIME_UTC` isn't necessarily monotonic which was kind of why I wanted to just drop it. I'm not completely sure but the timer is now always monotonic on the other platforms right? I figure it's better to just guarantee that for all usage `mp_time_ns`. AFAIK, this will technically break Solaris (dunno if mpv even works on that right now) but who cares?

Reply:
I'm fine with dropping it. We were using gettimeofday not so long ago, but if we stick to `clock_gettime` it is fine.

Reply:
@Dudemanguy: If it wasn't clear, I mostly meant the run-time fallback from `CLOCK_MONOTONIC_RAW` to `CLOCK_MONOTONIC`

Reply:
Oh, sure I suppose we can do that.